src/net/socket.rs

```diff
@@ -14,14 +14,39 @@ use std::{
     time::{Duration, Instant},
 };
 
+// Wrap `LinkConditioner` and `UdpSocket` together
+#[derive(Debug)]
+struct SocketWithConditioner {
+    socket: UdpSocket,
+    link_conditioner: Option<LinkConditioner>,
```

Comment:
can we make get methods here instead of directly accessing the struct internals

Reply:
Ok